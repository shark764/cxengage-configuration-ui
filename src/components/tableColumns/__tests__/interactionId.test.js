import interactionIdColumn, { interactionIdCell } from '../interactionId';

describe('InteractionId column', () => {
  it('returns the proper object configuration for the row', () => {
    expect(interactionIdColumn('0000-0000-0000-0001')).toMatchSnapshot();
  });
  it('renders the cell properly given a value', () => {
    expect(interactionIdCell('0000-0000-0000-0002')).toMatchSnapshot();
  });
  it('calling the columns Cell method with a value prop renders properly', () => {
    const cellObject = interactionIdColumn('0000-0000-0000-0001');
    expect(cellObject.Cell({ value: '0000-0000-0000-0003' })).toMatchSnapshot();
  });
});
