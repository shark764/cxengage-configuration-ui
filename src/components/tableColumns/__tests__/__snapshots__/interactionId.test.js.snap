// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InteractionId column calling the columns Cell method with a value prop renders properly 1`] = `
<span
  title="0000-0000-0000-0003"
>
  <CopyToClipboard
    onCopy={undefined}
    options={undefined}
    text="0000-0000-0000-0003"
  >
    <styled.div>
      0000-0000-0000-0003
    </styled.div>
  </CopyToClipboard>
</span>
`;

exports[`InteractionId column renders the cell properly given a value 1`] = `
<span
  title="0000-0000-0000-0002"
>
  <CopyToClipboard
    onCopy={undefined}
    options={undefined}
    text="0000-0000-0000-0002"
  >
    <styled.div>
      0000-0000-0000-0002
    </styled.div>
  </CopyToClipboard>
</span>
`;

exports[`InteractionId column returns the proper object configuration for the row 1`] = `
Object {
  "Cell": [Function],
  "Header": "Interaction Id",
  "accessor": "interactionId",
  "id": "interactionId",
  "show": "0000-0000-0000-0001",
}
`;
